packages/react-composites/review/beta/react-composites.api.md

```diff
@@ -602,6 +602,8 @@ export interface CallWithChatControlOptions {
     displayType?: CallControlDisplayType;
     endCallButton?: boolean;
     microphoneButton?: boolean;
+    // @beta
+    onFetchCustomButtonProps?: CustomCallControlButtonCallback[];
```

Comment:
I see `CustomCallControlButtonCallback` exposes the whole `ControlBarButtonProps` - I just want to confirm if we want to expose this whole interface or something more restrictive

Reply:
Actually `ControlBarButtonProps`'s interface doesn't quite work for the overflowMenu item - we may need to restrict it or have something like `(CustomCallControlButtonCallback | CustomCallControlOverflowButtonCallback)[]`

Reply:
This is a good point and would ultimately put the responsibility on Contoso to provide correct props for an overflow item. 
However, I've deliberately kept it as just customCallControlButtonCallback because I am automagically moving buttons into the overflow if certain conditions are met (max # of buttons reached, desktop to mobile view switched).
ControlBarButtonProps and DrawerMenuItemProps are similar enough that a DrawerMenuItem can be drawn from ControlBarButtonProps..

I am open to discuss how we would then automagically move buttons from the control bar to the overflow if we did not want to lean on converting ControlBarButtonProps and utilizing them as a DrawerMenuItemProps

Reply:
Thats a great point! With that we would still be able to extract and render a ControlBarButton and a DrawerMenuItem. 
Thanks James!